src/SemaphoreAirdrops.sol

```diff
@@ -0,0 +1,149 @@
+    constructor(
+        ISemaphore _semaphore
+    ) {
+        semaphore = _semaphore;
+    }
+
+    /// @notice Create a new airdrop
+    /// @param groupId The ID of the Semaphore group that will be elegible to claim this airdrop
+    /// @param token The ERC20 token that will be airdropped to elegible participants
+    /// @param holder The address holding the tokens that will be airdropped
+    /// @param amount The amount of tokens that each participant will receive upon claiming
+    function createAirdrop(
+        uint256 groupId,
+        ERC20 token,
+        address holder,
+        uint256 amount
+    ) public {
+        Airdrop memory airdrop = Airdrop({
+            groupId: groupId,
+            token: token,
+            manager: msg.sender,
+            holder: holder,
+            amount: amount
+        });
+
+        getAirdrop[airdropId] = airdrop;
+        emit AirdropCreated(airdropId++, airdrop);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                               CLAIM LOGIC                               ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Claim a given airdrop
+    /// @param airdropId The id of the airdrop getting claimed
+    /// @param receiver The address that will receive the tokens
+    /// @param root The of the Merkle tree
+    /// @param nullifierHash The nullifier for this proof, preventing double signaling
+    /// @param proof The zero knowledge proof that demostrates the claimer is part of the Semaphore group
+    function claim(
+        uint256 airdropId,
+        address receiver,
+        uint256 root,
+        uint256 nullifierHash,
+        uint256[8] calldata proof
+    ) public {
+        if (nullifierHashes[nullifierHash]) revert InvalidNullifier();
+
+        Airdrop memory airdrop = getAirdrop[airdropId];
+
+        semaphore.verifyProof(
+            root,
+            airdrop.groupId,
+            abi.encode(receiver).hashToField(),
+            nullifierHash,
+            abi.encode(address(this), airdropId).hashToField(),
+            proof
+        );
+
+        nullifierHashes[nullifierHash] = true;
+        emit AirdropClaimed(airdropId, receiver);
+
+        SafeTransferLib.safeTransferFrom(airdrop.token, airdrop.holder, receiver, airdrop.amount);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                               CONFIG LOGIC                             ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Update the details for a given airdrop, for addresses that haven't claimed already. Can only be called by the airdrop creator
+    /// @param airdropId The id of the airdrop to update
+    /// @param airdrop The new details for the airdrop
+    function updateDetails(uint256 airdropId, Airdrop calldata airdrop) public {
```

Comment:
Is there a recommended way to stop an Airdrop?

Reply:
you could set airdropAmount to 0 to prevent new tokens from getting delivered, or set groupId to 0 to make proofs fail, or just drain the holder to make transfers fail.